src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs

```diff
@@ -281,6 +279,18 @@ private static BoundExpression GetSpeculativelyBoundExpressionHelper(Binder bind
             return boundNode;
         }
 
+        protected static bool TryCreateSpeculativeFieldKeywordBinder(Binder binder, out SpeculativeFieldKeywordBinder speculativeFieldKeywordBinder)
```

Comment:
>TryCreateSpeculativeFieldKeywordBinder [](http://example.com/codeflow?start=30&length=38)

The "TryCreate" in the name implies that creation can fail. But we don't create the ```SpeculativeFieldKeywordBinder``` because it is not needed, not because something went wrong.
Consider renaming to something like "AddSpeculativeFieldKeywordBinderIfNeeded" and returning the original or ```SpeculativeFieldKeywordBinder```. This way all consumers will  be able to replace conditional logic around an assignment with unconditional assignment.